python/pyspark/sql/tests/test_types.py

```diff
@@ -1132,6 +1138,14 @@ def test_decimal_type(self):
         t3 = DecimalType(8)
         self.assertNotEqual(t2, t3)
 
+    def test_varchar_type(self):
+        v1 = VarcharType(10)
+        v2 = VarcharType(20)
+        self.assertTrue(v2 is not v1)
+        self.assertNotEqual(v1, v2)
+        v3 = VarcharType(10)
+        self.assertEqual(v1, v3)
```

Comment:
Shall we check if `v1 is v3`?

Reply:
`v1 is v3` should be True after parametric singleton is introduced. I will adjust that together in the followup.